custom_components/midea_ac_lan/config_flow.py

```diff
@@ -306,74 +302,72 @@ async def async_step_auto(
                 )
                 return await self.async_step_manually()
             # device not exist, get device detail from cloud
-            else:
-                if CONF_ACCOUNT not in self.account.keys():
-                    return await self.async_step_login()
-                if self.session is None:
-                    self.session = async_create_clientsession(self.hass)
-                if self.cloud is None:
+            if CONF_ACCOUNT not in self.account:
+                return await self.async_step_login()
+            if self.session is None:
+                self.session = async_create_clientsession(self.hass)
+            if self.cloud is None:
+                self.cloud = get_midea_cloud(
+                    self.account[CONF_SERVER],
+                    self.session,
+                    self.account[CONF_ACCOUNT],
+                    self.account[CONF_PASSWORD],
+                )
+            if not await self.cloud.login():
+                return await self.async_step_login()
+            self.found_device = {
+                CONF_DEVICE_ID: device_id,
+                CONF_TYPE: device.get(CONF_TYPE),
+                CONF_PROTOCOL: device.get(CONF_PROTOCOL),
+                CONF_IP_ADDRESS: device.get(CONF_IP_ADDRESS),
+                CONF_PORT: device.get(CONF_PORT),
+                CONF_MODEL: device.get(CONF_MODEL),
+            }
+            if device_info := await self.cloud.get_device_info(device_id):
+                # set subtype with model_number
+                self.found_device[CONF_NAME] = device_info.get("name")
+                self.found_device[CONF_SUBTYPE] = device_info.get("model_number")
+            # get token and key from cloud for v3 device
+            if device.get(CONF_PROTOCOL) == 3:
+                if self.account[CONF_SERVER] == "美的美居":
+                    _LOGGER.debug(
+                        "Try to get the Token and the Key use the preset MSmartHome account",
+                    )
                     self.cloud = get_midea_cloud(
-                        self.account[CONF_SERVER],
+                        "MSmartHome",
```

Comment:
Can you explain why we need to hardcore a value here ?

Reply:
This is a display bug of `git diff`, here only reduce the indentation